Sys/TomPIT.Sys/Workers/ScheduleCalculator.cs

```diff
@@ -226,20 +245,37 @@ private static DateTime CalcNextRunYearly(IScheduledJob worker, DateTime initial
 				initial = CorrectStart(worker, initial, now);
 				var nextRun = FixTime(worker, initial);
 
-				nextRun = new DateTime(nextRun.Year + (worker.IntervalValue == 0 ? 1 : worker.IntervalValue), 1, 1, nextRun.Hour, nextRun.Minute, nextRun.Second);
+                var intervalValue = IntervalValue < 1 ? 0 : IntervalValue;
+
+                nextRun = new DateTime(nextRun.Year, 1, 1, nextRun.Hour, nextRun.Minute, nextRun.Second);
 
 				switch (worker.YearMode)
 				{
 					case WorkerYearMode.ExactDate:
-						return EnsureValidDate(worker, new DateTime(nextRun.Year, worker.MonthNumber, worker.DayOfMonth, nextRun.Hour, nextRun.Minute, nextRun.Second));
+                        nextRun = EnsureValidDate(worker, new DateTime(nextRun.Year, worker.MonthNumber, worker.DayOfMonth, nextRun.Hour, nextRun.Minute, nextRun.Second));
+						break;
 					case WorkerYearMode.RelativeDate:
 						nextRun = new DateTime(nextRun.Year, worker.MonthNumber, 1, nextRun.Hour, nextRun.Minute, nextRun.Second);
-
-						return ProcessMonthPart(worker, nextRun, WorkerMonthMode.RelativeDay);
+                        nextRun ProcessMonthPart(worker, nextRun, WorkerMonthMode.RelativeDay);
```

Comment:
Is this line missing an equals character between _nextRun_ and _ProcessMonthPart_?

Reply:
Yes. Added to next commit.